### PR TITLE
fix ls with empty string

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description
Fixes: #12054

It's cause by nu always add `/*` if there is a parameter in ls, then `ls ""` becomes `ls "/*"`.  This pr tries to fix it by only append `/` character if pattern is not empty.

# User-Facing Changes
NaN

# Tests + Formatting
Done

# After Submitting
NaN

### CODE REVIEW

File: crates/nu-command/src/filesystem/ls.rs [hunk 1]
```diff
@@ -90,6 +90,13 @@ impl Command for Ls {
         let pattern_arg = opt_for_glob_pattern(engine_state, stack, call, 0)?;
         let pattern_arg = {
             if let Some(path) = pattern_arg {
+                // it makes no sense to list an empty string.
+                if path.item.as_ref().is_empty() {
+                    return Err(ShellError::FileNotFoundCustom {
+                        msg: "empty string('') file does not exist".to_string(),
```
Comment (sholderbach) [inline crates/nu-command/src/filesystem/ls.rs:96]: would it be more likely that you expect to use expect to use it for a directory?
```suggestion
                        msg: "empty string('') directory or file does not exist".to_string(),
```

I don't know my platforms well enough if any system allows an empty string file/directory name at all.

Comment (WindSoilder) [reply crates/nu-command/src/filesystem/ls.rs]: I've tried on both linux/mac/windows, and they don't allow empty string file, with different error message.

Comment (sholderbach) [reply crates/nu-command/src/filesystem/ls.rs]: OK perfect! Let's land this :)

File: crates/nu-command/src/filesystem/ls.rs
Comment (bbbiyeba) [inline crates/nu-command/src/filesystem/ls.rs:1]: Code looks clean, but logic could be simplified ( the inner block handles the Some case and falls through to else { pattern_arg } which just returns the original None )

File: crates/nu-command/tests/commands/ls.rs
Comment (bbbiyeba) [inline crates/nu-command/tests/commands/ls.rs:1]: No Issues here

**Pull request merged**